### PR TITLE
Eltek Support

### PR DESCRIPTION
**Pull request by sippe2**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: includes/polling/os/eltek.inc.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2018 Mikael Sipilainen
+ * @author     Mikael Sipilainen <mikael.sipilainen@gmail.com>
+ */
+
+$version   = snmp_get($device, '.1.3.6.1.2.1.1.1.0', '-OQva', 'SNMPv2-MIB');
```
Comment (murrant) [inline includes/polling/os/eltek.inc.php]: You don't need to re-fetch this, just use:
```php
$version = $device['sysDescr'];
```

**Comment by murrant** [comment]
Thanks for the contribution.  You could probably update them in this file.

File: includes/polling/os/eltek.inc.php (hunk 1 above)
Comment (sippe2) [reply includes/polling/os/eltek.inc.php]: Hopefully updated version looks better now.

**Comment by laf** [comment]
> Just wondering is there some clever way to get sysname, sysLocation and sysContact information from different oid? Eltek stores that kind of data to different place...

This seems like it's better to contact the vendor and ask them to implement the standard MIBs properly.

Got a link to the vendor / devices this is for please?

You also need to submit tests for this: https://docs.librenms.org/#Developing/os/Test-Units/#example-workflow

**Comment by sippe2** [comment]
I'll try to make those tests. Only problem is snmpbulkwalk jams totally and snmpwalk complain 'oid not increasing' without -Cc command. So I have some understanding problems to make correct like snmprec query for snmpsim. Oid 1.3.6.1.4.1.12148.9 works perfectly but standard oid 1.2.1.1.0 not increasing after 1.2.7.5.1.1.0.0.0.0.4002...

**Comment by murrant** [comment]
@sippe2 If you can just run `./scripts/collect-snmp-data.php -h HOSTNAME` and add the snmprec to this PR we can generate the test data.

If you would like to get snmpsim working, remove any currently installed snmpsim.  Then run `pip3 install snmpsim`.

**Comment by murrant** [comment]
I was about to add the json file, when I noticed a couple things:
This OS should be caled webpower.
The version string should be just the version.
You should not jam unrelated data into the features string.

**Comment by sippe2** [comment]
Ok.
So you suggest that I rename eltek to webpower (lowercase I suppose).
Did I understood correctly that the version should be just number like "4.11"?

Sorry about the confusion. The first time ever publishing any kind of code and I can see there is a lot of things to know. :)

**Comment by sippe2** [comment]
Actually looks there is already os called webpower but different oid.

**Comment by murrant** [comment]
There is indeed, how about eltek-webpower?

**Comment by sippe2** [comment]
Hopefully looks better now :)

File: includes/polling/os/eltek-webpower.inc.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+// ELTEK-DISTRIBUTED-MIB (version9) needs files SNMPv2-SMI SNMPv2-TC
+
+$output = preg_split("/[\s,]+/", $device['sysDescr']);
+        $version = $output[1];
+
+$oid = snmp_get_multi($device, 'systemSiteInfoControllerType.0 rectifierStatusType.1 batteryName.0 systemSiteInfoSystemSeriaNum.0 systemSiteInfoCustomer.0 systemSiteInfoLocation.0', '-OQUs', 'ELTEK-DISTRIBUTED-MIB');
+        $hardware = $oid[0]['systemSiteInfoControllerType'];
+        $features = 'Rectifier type: '.$oid[1]['rectifierStatusType'];
+        $features .= ', Battery name: '.$oid[0]['batteryName'];
+        $features .= ', Site Customer: '.$oid[0]['systemSiteInfoCustomer'];
```
Comment (murrant) [inline includes/polling/os/eltek-webpower.inc.php]: You still need to remove these extra things from features.

**Comment by sippe2** [comment]
Travis CI broken? I think it jammed during the tests...

**Comment by murrant** [comment]
No, test data wasn't added yet, because of pending changes.  I've added it now.

**Comment by sippe2** [comment]
Thanks for the help. :)

**Comment by murrant** [comment]
Thanks for the contribution :)

**Comment by murrant** [comment]
@sippe2 Hey, if these support UPS-MIB instead of doing what you have done, you just need to add `rfc1628_compat: true` to the os yaml file.

**Comment by murrant** [comment]
@sippe2 looks like you have accidentally added some netagent2 changes to this PR.

**Comment by sippe2** [comment]
Sigh... Hard to learn how git got works. Actually I just did not understood to create a new branch... and I just pushed new files here. Svn was easy but it was 10+ years ago. I'll try fo fix my mess. My bad. Sorry.

**Comment by murrant** [comment]
https://docs.librenms.org/Developing/Using-Git/

**Comment by sippe2** [comment]
Tavis CI says...

FAILURES!
Tests: 492, Assertions: 1246, Failures: 1.
Cleaning database...
snmpsimd: output at /tmp/snmpsimd.log
The command "php scripts/pre-commit.php -q -u --db --snmpsim --fail-fast" exited with 1.
0.01s$ bash -n daily.sh

Just wondering is it better to start over and create a new pull request in the new branch with all files what needed?

**Comment by murrant** [comment]
Up to you, I can't help you out because it is on your master branch :)

**Pull request closed without merging**